docs/src/tcp.rst

```diff
@@ -75,11 +75,11 @@ API
     Bind the handle to an address and port. `addr` should point to an
     initialized ``struct sockaddr_in`` or ``struct sockaddr_in6``.
 
-    When the port is already taken, you can expect to see an ``UV_EADDRINUSE``
-    error from either :c:func:`uv_tcp_bind`, :c:func:`uv_listen` or
-    :c:func:`uv_tcp_connect`. That is, a successful call to this function does
-    not guarantee that the call to :c:func:`uv_listen` or :c:func:`uv_tcp_connect`
-    will succeed as well.
+    When the port is already taken, the error is relayed and you can expect to see an 
+    ``UV_EADDRINUSE`` error from function such as :c:func:`uv_listen` or 
```

Comment:
Yes, deferred is a better choice. 
preliminary code scan shows that delayed_error is being returned at
https://github.com/deleisha/libuv/blob/v1.x/src/unix/tcp.c#L212
https://github.com/deleisha/libuv/blob/v1.x/src/unix/tcp.c#L234

Hence, My thinking was that function like `uv_tcp_getpeername` and `uv_tcp_getsockname`.